presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java

```diff
@@ -0,0 +1,258 @@
+                int columnIndex = -20;
+                HiveType hiveType = HiveType.toHiveType(hiveTypeTranslator, callExpression.getType());
+                if (!callExpression.getArguments().isEmpty()) {
+                    RowExpression column = callExpression.getArguments().get(0);
+                    colName = column.toString();
+                    HiveColumnHandle oldColumnHandle = (HiveColumnHandle) oldTableScanNode.getAssignments().get(column);
+                    columnIndex = oldColumnHandle.getHiveColumnIndex();
+                    hiveType = oldColumnHandle.getHiveType();
+                }
+
+                ColumnHandle newColumnHandle = new HiveColumnHandle(
+                        colName,
+                        hiveType,
+                        callExpression.getType().getTypeSignature(),
+                        columnIndex,
+                        HiveColumnHandle.ColumnType.AGGREGATED,
+                        Optional.of("partial aggregation pushed down"),
+                        Optional.of(aggregationEntry.getValue()));
+                assignments.put(aggregationEntry.getKey(), newColumnHandle);
+            }
+
+            HiveTableLayoutHandle oldTableLayoutHandle = (HiveTableLayoutHandle) oldTableHandle.getLayout().get();
+            HiveTableLayoutHandle newTableLayoutHandle = new HiveTableLayoutHandle(oldTableLayoutHandle.getSchemaTableName(),
+                    oldTableLayoutHandle.getPartitionColumns(),
+                    oldTableLayoutHandle.getDataColumns(),
+                    oldTableLayoutHandle.getTableParameters(),
+                    oldTableLayoutHandle.getPartitions().get(),
+                    oldTableLayoutHandle.getDomainPredicate(),
+                    oldTableLayoutHandle.getRemainingPredicate(),
+                    oldTableLayoutHandle.getPredicateColumns(),
+                    oldTableLayoutHandle.getPartitionColumnPredicate(),
+                    oldTableLayoutHandle.getBucketHandle(),
+                    oldTableLayoutHandle.getBucketFilter(),
+                    oldTableLayoutHandle.isPushdownFilterEnabled(),
+                    oldTableLayoutHandle.getLayoutString(),
+                    oldTableLayoutHandle.getRequestedColumns(),
+                    true);
+
+            TableHandle newTableHandle = new TableHandle(
+                    oldTableHandle.getConnectorId(),
+                    hiveTableHandle,
+                    oldTableHandle.getTransaction(),
+                    Optional.of(newTableLayoutHandle));
+
+            return Optional.of(new
+
+                    TableScanNode(
+                    idAllocator.getNextId(),
+                    newTableHandle,
+                    ImmutableList.copyOf(partialAggregationNode.getOutputVariables()),
+                    ImmutableMap.copyOf(assignments),
+                    oldTableScanNode.getCurrentConstraint(),
+                    oldTableScanNode.getEnforcedConstraint()));
```

Comment:
We don't have to just return TableScanNode. We could still have the AggNode -> TableScanNode to make the optimizer doesn't fail. This assumption holds given the input type for min/max is the same for its output.

Reply:
As we discussed elsewhere, this would work for min/max - but we would lose the functionality of realizing aggregate pushdown for the two flavors of count (count(*) and count(column name)) that is currently implemented in this PR. Is it worth it to maintain this invariant, given the potential loss of functionality?

Reply:
Let's have it at the moment. BTW, ORC min/max is not tied. They are not necessarily the tight upper or lower bound of the data. With this, potentially, we will have correctness issues. Have you thought about that?

Reply:
Added a new session parameter to control partial aggregation pushdown for variable length types to handle truncation/overflow